baseball/js/app.js

```diff
@@ -0,0 +1,366 @@
+			
+			//Draw Cities
+			map.selectAll('circle')
+				.data(appData.stadiums)
+				.enter()
+				.append('circle')
+				.classed('stadium', true)
+				.attr('cx', function(d){
+					return projection([d.lon, d.lat])[0];
+				})
+				.attr('cy', function(d){
+					return projection([d.lon, d.lat])[1];
+				})
+				.attr('r', function(d){
+					var radius = d.years[yearSelectValue][playerSelectValue] * 100;
+					return findRadius(radius);
+				})
+				.style('fill', function(d){
+					var radius = d.years[yearSelectValue][playerSelectValue] * 100;
+					return color(findRadius(radius));
+				})
+				.style('opacity', 0.8)
+				.style('cursor', 'pointer')
+				.text(function(d){
+					return d.years[yearSelectValue][playerSelectValue];
+				})
+				.attr('font-size', '14px')
+				.attr('font-weight', 'bold')
+				.attr('fill', '#000')
+				.on('mouseover', function(d){
+				
+					//set stadium name
+					d3.select('.mapTooltip h3').text(function(){
+						return d.name;
+					});
+
+					//set location
+					d3.select('.mapTooltip .location').text(function(){
+						return d.location;
+					});
+
+					//set teamname
+					d3.select('.mapTooltip .teamName').text(function(){
+						return d.teamName;
+					});
+
+					//set capacity
+					d3.select('.mapTooltip .capacity').text(function(){
+						return d.capicity;
+					});
+				
+					//Set batting Average
+					d3.select('.mapTooltip .battingAverage').text(function(){
+						return setbattingAverage(d);
+					});
+
+					d3.select(".mapTooltip")
+						.style("display", "block")
+						.style("left", function() { // stop tooltip from being cutoff at right of page
+						  var right_edge = d3.event.pageX + this.offsetWidth; //this.offsetWidth == width of tooltip
+						  if (right_edge > width) {
+							return (d3.event.pageX - this.offsetWidth)+"px";
+						  }
+						  return (d3.event.pageX)+"px";
+						})
+						.style("top", function() { // stop tooltip from being cutoff at bottom of page
+						  var bottom_edge = d3.event.pageY + this.offsetHeight; //this.offsetHeight == height of tooltip
+						  if (bottom_edge > height) {
+							return (d3.event.pageY - this.offsetHeight)+"px";
+						  }
+						  return (d3.event.pageY)+"px";
+						});
+			})
+			.on("mouseout", function() {
+			  d3.select(".mapTooltip").style("display", "none");
+			});
+			
+			//Create Legend
+			var legend = legendSVG.selectAll('.legendKey')
+				.data(color.domain())
+				.enter()
+				.append('g')
+				.classed('legendKey', true)
+				.attr('transform', function(d, i){
+					var width = legendSquareSize;
+					var horz = i * width;
+					return 'translate(' + horz + ',5)';
+				});
+				
+			legend.append('rect')
+				.attr('width', legendSquareSize)
+				.attr('height', 10)
+				.style('fill', color);
+			
+			legend.append('text')
+				.attr('x', 20)
+				.attr('y', 25)
+				.style("text-align", "center")
+				.style("font-size", ".8em")
+				.text(function(d){
+					if(d === 5){
```

Comment:
this function could be written as a switch statement if you wanted; it'd be ever slightly more concise. totally optional, though. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
So 
```
switch (expr) {
   case '5': '< .250'
}
```
?

Reply:
i think no quotes around the 5, and all the cases:
```js
.text(function(d) {
  var label;
  switch(d) {
    case 5: label = '< .250'; break;
    case 10: label = '< .300'; break;
    case 20: label = '< .400'; break;
    case 30: label = '< .700'; break;
    case 40: label = '> .700'; break;
    default: label = '';
  };
  return label;
});
but now that i've tried it out, i'm not sure it actually saves any characters over a bunch of `if` statements in javascript...

Reply:
I guess you could also do it with an object, but then you have to convert to character with `''+d`, which is also ugly:
```
labels = {
  5: '< .250',
  10: '< .300',
  20: '< .400',
  30: '< .700',
  40: '> .700'
};
labels[d]
```

Reply:
It does have some readability ability over a bunch of if and else ifs, but yeah not much on the character count saving. lol